### DISCUSSION TITLE
Lost all my requests and collections after the new update (2022.7.0)

### DISCUSSION THREAD

**Discussion by tkurskiev** [Kong/insomnia#5538] [Feedback]
I'm very angry and I delete this app. This is bullshit, guys.

Yeah, I was able to find them in "C:\Users\User_name\AppData\Roaming\Insomnia\insomnia.Request.db", but this is very unpleasant experience.

**Comment by timoheutsrs** [comment]
How did you managed to import the data again? I keep getting errors when opening that file

**Reply by tkurskiev** [reply]
I opened it in Notepad++, analysed and copied what I need by hand into Postman

**Comment by inafil76** [comment]
After the 2022.7.0 update, I keep getting errors when I open my Collections. They are now labeled as Documents and the error says that they don't follow the correct format for a document.

Tried exporting and importing again but it seems that importing has issues also.
This really takes me back, I had a LOT of work in those collections...

**Comment by thayaneBomfims** [comment]
Concordo completamente. Perdi todas as requisições com essa atualização ridícula. Todos me dizem para utilizar o Postman e eu sempre relutei, mas estão certos. Esse app é um lixo mesmo, vivo tendo problemas

**Comment by aidarchikable** [comment]
I lost all my collections and environments because of the last update.
And I'm very disappointed because I used Insomnia as a replacement for Postman, but now Insomnia has become Postman.

**Comment by josejesusrl** [comment]
La verdad no me gustó la actualización, me gustaría regresar a una versión anterior o buscar un nuevo cliente rest

**Comment by TheKayThatWasOrange** [comment]
It's funny how they keep doing this over and over and keep getting angry at users "spamming" their issues with complaints about how completely irresponsible they are.

**Comment by subnetmarco** [comment]
Since a very long time Insomnia has had weird edge cases to implement flawless migrations across different types of data structures across versions. We are taking action and have fully allocated resources to finally build reliable migrations and try to eliminate these edge cases, which were caused by very old code.

These migrations happen on the file system, and needed more resourcing to properly handle all sort of edge cases that happen when interacting with the underlying I/O, failures and so on.

As such in v8.3 we shipped many of those improvements already, and in v8.4 we are shipping another large set of improvements with better error handling in those circumstances.

In the meantime, I offer one of our Insomnia engineers to hop on a Zoom call to debug and look at the migration problems you may experience. We are literally collecting all the edge cases as we speak via many of these calls.

@TheKayThatWasOrange Complaining about legitimate issues is **not** spamming. But actual spamming is..spamming, and also in violation of the terms of service of GitHub.

**Reply by TheKayThatWasOrange** [reply]
Maybe you can go throw another tantrum on HN about it.

**Comment by subnetmarco** [comment]
Closing this discussion since we have implemented several improvements and we are not seeing this issue popping up anymore. I will re-open it if the issue persists.

**Reply by gian1200** [reply]
Hello @subnetmarco. I just updated from 2023.5.8 to [12.3.1](https://insomnia.rest/changelog#12.3.1), and I lost my requests/Collections.

As mentioned above, I think I found my requests at "C:\Users\User_name\AppData\Roaming\Insomnia\insomnia.Request.db".  (as well as other DB files)

By any chance, do you know if there is any tool that I can use to reimport everything to the just installed version? Is this supported by Insomnia in some way?

Doing some googling at the moment without success.

PS: would be nice if also collections/certificates/preferences are imported, too (desired, not required)

Thanks!

**Reply by subnetmarco** [reply]
@chayglass84

**Reply by chayglass84** [reply]
Let me look into this, back soon.

Is it possible you signed in with a different email address or account, even slightly? That would explain why it all still exists but you can't see it inside Insomnia.

**Reply by gian1200** [reply]
Hi, thanks for your answer. Actually, I wasn't logged in. I haven't been for the last few months/years.
